mne/beamformer/_compute_beamformer.py

```diff
@@ -438,3 +445,99 @@ def _compute_beamformer(method, G, Cm, reg, n_orient, weight_norm,
                 W /= norm[:, np.newaxis]
 
     return W, is_free_ori
+
+
+class Beamformer(dict):
+    """A computed beamformer.
+
+    Notes
+    -----
+    .. versionadded:: 0.17
+    """
+
+    def copy(self):
+        """Copy the beamformer.
+
+        Returns
+        -------
+        beamformer : instance of Beamformer
+            A deep copy of the beamformer.
+        """
+        return deepcopy(self)
+
+    def __repr__(self):  # noqa: D105
+        n_verts = sum(len(v) for v in self['vertices'])
+        n_channels = len(self['ch_names'])
+        if self['subject'] is None:
+            subject = 'unknown'
+        else:
+            subject = '"%s"' % (self['subject'],)
+        out = ('<Beamformer : %s : subject %s : %s vert, %s ch'
+               % (self['kind'], subject, n_verts, n_channels))
+        if self['pick_ori'] is not None:
+            out += ' : %s ori' % (self['pick_ori'],)
+        if self['weight_norm'] is not None:
+            out += ' : %s norm' % (self['weight_norm'],)
+        if self.get('inversion') is not None:
+            out += ' : %s inversion' % (self['inversion'],)
+        if 'rank' in self:
+            out += ' (rank %s)' % (self['rank'],)
+        out += ' >'
+        return out
+
+    @verbose
+    def save(self, fname, overwrite=False, verbose=None):
+        """Save the beamformer filter.
+
+        Parameters
+        ----------
+        fname : str
+            The filename to use to write the HDF5 data.
+            Should end in ``'-lcmv.h5'`` or ``'-dics.h5'``.
+        overwrite : bool
+            If True, overwrite the file (if it exists).
+        verbose : bool, str, int, or None
+            If not None, override default verbose level (see
+            :func:`mne.verbose` and :ref:`Logging documentation <tut_logging>`
+            for more).
+        """
+        ending = '-%s.h5' % (self['kind'].lower(),)
+        check_fname(fname, self['kind'], (ending,))
+        csd_orig = None
+        try:
+            if 'csd' in self:
+                csd_orig = self['csd']
+                self['csd'] = self['csd'].__getstate__()
```

Comment:
would be cool if `h5io` could do this automatically.

Reply:
Maybe eventually, but let's not do it in this PR. It will take a bit of thinking and decisions, such as how to deal with reading end of things (re-instantiating the class, which may or may not always be possible depending on imports, the class `__init__` itself, etc.).